Added Functionality : Sword and Basilisk(Monster)

saarthakchats opened the pull request:
<!-- Don't delete anything without explicit instructions from a maintainer. -->

<!-- Check by changing each `[ ]` to `[x]` Please take note of the whitespace as it matters. -->


### Changes done in this Pull Request


- <!-- If you fully fixed/improved some isuue(s), please insert the issue number(s) behind the # 
       If you have not fixed some issue(s) completly, but only some of step(s) in issue(s),
         please insert the issue number(s) with step number(s) hind the # -->
  Related Issues or Pull Requests #<Issue no.>


<!-- please summarize the problem you faced -->
<!-- Please remove unwanted words in following topic -->
### The problem I want to solve or the facility I want to improve is/are
<!-- Mention the bug/facility solved/improved -->

### My steps to solve it
<!-- Please summarize the solution you chose.
     Mention the files changed. Add what changes you have done. -->

### Screenshots,

saarthakchats commented:
@niccokunzmann Pls merge :D

niccokunzmann commented on js/tiles.js:
```diff
@@ -974,11 +983,39 @@ const door = {
+        visit: function() {
+             if (player.inventory.has('Sword')) {
+             alertSuccess("You killed the Monster! Yay!!");
+             player.inventory.remove('Spoon');
```
why do you remove the spoon?

niccokunzmann replied:
How about is drops some item, just an idea, nothing blocking.

niccokunzmann commented on js/tiles.js:
```diff
@@ -974,11 +983,39 @@ const door = {
         }
     }),
     threeHeads: Object.assign({}, OpenDoors, {
+    	canEnterFromTheRight() {
```
You can change all these functions but please also change the look. Currently, it can be entered from left, top, bottom.

niccokunzmann replied:
@saarthakchats Please respond to this. I just want your view on this.

saarthakchats commented:
@niccokunzmann Done changes

Beta-King commented:
please give a preview link :D

saarthakchats commented:
@Beta-King 
The game is currently down

Beta-King commented:
Oh is that so ?? kk nevermind then, ill check it in code basis :D

saarthakchats commented:
Yes please :D

Beta-King commented:
It doesnt look good, could you change that ??

Beta-King commented on js/tiles.js:
```diff
@@ -974,11 +983,39 @@ const door = {
+             alertSuccess("You killed the Monster! Yay!!");
+             player.inventory.remove('Sword');
+            } else {
+                alertNormal("You Need a Sword to fight the Basilisk!", "It is deadly");
```
this statement has no need as the player cant enter the tile anyways, all entries are locked until you get a sword right ?? so i suggest you open the entrance and design a badge and the player will only get the badge if he kills the monster, don't you think that'll make more sense ??

saarthakchats commented:
@niccokunzmann 
I have made the changes you asked for

niccokunzmann commented:
The game should work. At least the master branch works for me.

saarthakchats commented:
Ya, can we merge

saarthakchats commented:
@niccokunzmann

niccokunzmann commented:
@saarthakchats I would like to ask you again, in case you did overlook this: 
There are a lot of forbidden entering directions. These should match the design.

- Is it really your intention to [show a door](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR1005) when you can not enter or [leave to the top](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR998) ?
- You do not show a wall on the left, yet [you forbid entering and leaving there](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR992) - is this really your intention?

If you can both answer with yes, I will merge and we create an issue to discuss this there. People who disagree might want to revert if they do not like to fix the code. 
Please answer these two questions, so I am sure this is considered.

niccokunzmann commented:
There is one blocker: Please show that we have the rights to use the basilisk image.

saarthakchats commented:
Taken from pixabay, added to CCO license folder in Images from Pixabay

saarthakchats commented:
@niccokunzmann

niccokunzmann commented:
Okay once again: Please answer to these two questions with yes and I will merge:


- Is it really your intention to [show a door](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR1005) when you can not enter or [leave to the top](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR998) ?
- You do not show a wall on the left, yet [you forbid entering and leaving there](https://github.com/fossasia/labyrinth/pull/495/files#diff-5be326fac7a1cad78493e00680aab52dR992) - is this really your intention?

See https://github.com/fossasia/labyrinth/pull/495#issuecomment-357505949

saarthakchats commented:
Yes

saarthakchats commented:
We can create an issue after this if you want

saarthakchats commented:
@niccokunzmann

niccokunzmann commented:
So, you license the image under AGPL and under CC0. Okay. Please move the tile to that folder and license the image in the tile under CC0 and then remove the png file.

niccokunzmann commented:
@saarthakchats Please create the issue.

saarthakchats commented:
Can you give details of the issue

saarthakchats commented:
@niccokunzmann

niccokunzmann commented:
I need time to digest this conversation. Please ask in the chat what other people think.

saarthakchats commented:
@niccokunzmann 
I have removed the basilisk image and added a snake drawn on inkscape, so the licensing issue doesnt exist anymore, Can we merge now?

saarthakchats commented:
Is it okay now?
@niccokunzmann

niccokunzmann commented on js/tiles.js:
```diff
@@ -974,11 +983,38 @@ const door = {
+        canLeaveToTheRight() {
+            return false;
+        },
+        canEnterFromTheLeft() {
```
please remove lines 992 to 996 to allow entering an leaving on the left.

niccokunzmann commented on js/tiles.js:
```diff
@@ -974,11 +983,38 @@ const door = {
+        canLeaveToTheLeft() {
+            return false;
+        },
+        canEnterFromTheTop: function(player) {
```
please remove lines 997 to 1002 to allow entering and leaving at the top.

niccokunzmann commented on js/tiles.js:
```diff
@@ -974,11 +983,38 @@ const door = {
+        },
+        visit: function() {
+             if (player.inventory.has('Sword')) {
+             alertSuccess("You killed the Monster! Yay!!");
```
Maybe you can say here that the sword got stuck ^^

saarthakchats replied:
In line 1010?

saarthakchats commented:
@niccokunzmann 
Sorry I left that, done now

niccokunzmann commented:
@saarthakchats Thanks, it looks good to me now. I merge. Please test after this if all is as expected.

saarthakchats commented:
Ok thanks for merging
